wavenet/model.py

```diff
@@ -84,8 +84,17 @@ def __init__(self,
         self.initial_filter_width = initial_filter_width
         self.histograms = histograms
 
+        self.receptive_field = self._calculate_receptive_field()
         self.variables = self._create_variables()
 
+    def _calculate_receptive_field(self):
+        receptive_field = (self.filter_width - 1) * sum(self.dilations) + 1
+        if self.scalar_input:
+            receptive_field += self.initial_filter_width - 1
+        else:
+            receptive_field += self.filter_width - 1
+        return receptive_field
```

Comment:
I think you're forgetting about the initial causal convolution (line 313 in model.py):
`        current_layer = self._create_causal_layer(current_layer)`
it has dilation 1 and width equal to filter_width when scalar_input=False, so in your example it contributes 1 to the receptive field.
Also, there is a runtime check here (line 525 in model.py):
`                loss = tf.nn.softmax_cross_entropy_with_logits(
                    prediction,
                    target_output)
`
this line will fail if the receptive field is calculated incorrectly because of dimension mismatch.
Your implementation might also be correct, but it assumes a specific form of the dilation stack (1, 2, 4, ..., 2^n, 1, 2, 4, ..., 2^n, ...) and mine doesn't.